sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs

```diff
@@ -60,7 +60,8 @@ public static void ReportDiagnosticWhenActive(this CodeBlockAnalysisContext cont
 
         private static void ReportWhenNotSuppressed(SyntaxTree tree, Diagnostic diagnostic, Action<Diagnostic> report)

```

Comment:
This method name is misleading now that more than the suppression state being taken into account.

Reply:
WDYT of `TryReportDiagnostic`?